### ISSUE TITLE
org-tree-slide-mode bugs

### ISSUE THREAD

**Issue by ngharrison** [doomemacs/doomemacs#4646]
**What did you expect to happen?**
Be able to step through my headings and subheadings forwards and backwards, showing images, and prettified.

**What actually happened?**
Content under subheadings is hidden and only the first subheading is shown. All the rest get skipped when moving to the next slide. I also can't move backwards in the slides, only forwards.
![image](https://user-images.githubusercontent.com/7918794/107832639-32ace380-6d4e-11eb-9f24-d9c6221e55b1.png)

I can manually open a heading, but then no images are shown, no `#+` blocks are removed and no symbols are prettified. It actually un-prettifies things if I have that as a setting before starting the mode.
![image](https://user-images.githubusercontent.com/7918794/107832851-bebf0b00-6d4e-11eb-8457-5a69f19c5fd6.png)

**Additional details:**
- I have (org +present) in my init.el
- I've tried combinations of running without my personal config and using the example init file, but those made no difference.
- Removing the config in ~/.emacs.d/modules/lang/org/contrib/present.el does revert the behavior back to something useable. This points to it definitely being the culprit.

**Steps to reproduce:**
1. I have (org +present) in my init.el
2. Open an org file with headings and subheadings and content
3. Run `M-x org-tree-slide-mode`
4. Optionally try moving between slides with `C-right` and `C-left`

**System information:**
<details><pre>
```
SYSTEM  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /bin/bash
        uname      Linux 5.4.0-65-generic #73-Ubuntu SMP Mon Jan 18 17:25:17 UTC 2021 x86_64
        path       (~/miniconda3/bin/ ~/.emacs.d/bin/ ~/.emacs.d/bin/ ~/.emacs.d/bin/ ~/miniconda3/bin ~/.emacs.d/bin/ ~/miniconda3/bin ~/.emacs.d/bin/ ~/miniconda3/bin ~/.emacs.d/bin/ ~/miniconda3/bin ~/.emacs.d/bin/ ~/miniconda3/bin ~/.emacs.d/bin/ ~/miniconda3/bin ~/.emacs.d/bin ~/.cargo/bin ~/.emacs.d/bin ~/miniconda3/bin ~/.emacs.d/bin ~/miniconda3/bin ~/.emacs.d/bin ~/miniconda3/bin ~/.emacs.d/bin ~/miniconda3/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/.emacs.d/bin ~/miniconda3/condabin ~/.emacs.d/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /snap/bin /usr/lib/emacs/27.1/x86_64-linux-gnu/)
EMACS   dir        ~/.emacs.d/
        version    27.1
        build      Sep 19, 2020
        buildopts  --build=x86_64-linux-gnu --prefix=/usr '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' --sysconfdir=/etc --localstatedir=/var --disable-silent-rules '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' --disable-maintainer-mode --disable-dependency-tracking --prefix=/usr --sharedstatedir=/var/lib --libexecdir=/usr/lib --localstatedir=/var/lib --infodir=/usr/share/info --mandir=/usr/share/man --enable-locallisppath=/etc/emacs:/usr/local/share/emacs/27.1/site-lisp:/usr/local/share/emacs/site-lisp:/usr/share/emacs/27.1/site-lisp:/usr/share/emacs/site-lisp --program-suffix=27 --with-modules --with-file-notification=inotify --with-mailutils --with-harfbuzz --with-json --with-x=yes --with-x-toolkit=gtk3 --with-lcms2 --with-cairo --with-xpm=yes --with-gif=yes --with-gnutls=yes --with-jpeg=yes --with-png=yes --with-tiff=yes --with-xwidgets 'CFLAGS=-g -O2 -fdebug-prefix-map=/build/emacs27-bifpWT/emacs27-27.1~1.git86d8d76aa3=. -fstack-protector-strong -Wformat -Werror=format-security -no-pie' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -no-pie'
        features   XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND GPM DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD JSON PDUMPER LCMS2 GMP
        traits     (gui server-running envvar-file)
DOOM    dir        ~/.doom.d/
        version    2.0.9
        font       #<font-spec nil nil Hasklig nil nil nil nil nil 14 nil nil nil nil>
        theme      doom-oceanic-next
        build      HEAD -> develop 3a7be7bb4 2021-02-01 02:45:45 -0500
        elc-files  0
        modules    (:completion (company +childframe) (ivy +prescient +icons) :ui doom doom-dashboard hl-todo hydra minimap modeline ophints (popup +defaults) treemacs vc-gutter vi-tilde-fringe (window-select +numbers) workspaces zen :editor (evil +everywhere) file-templates fold multiple-cursors rotate-text snippets word-wrap :emacs (dired +icons) electric (ibuffer +icons) undo vc :term eshell vterm :checkers syntax (spell +aspell) grammar :tools biblio (eval +overlay) (lookup +dictionary) magit pdf :lang data emacs-lisp julia kotlin latex markdown (org +jupyter +roam +present +noter) (python +conda) sh yaml :config literate (default +bindings +smartparens))
        packages   ((sublimity :recipe (:type git :host github :repo ngharrison/sublimity)) (centered-cursor-mode) (vimrc-mode))
        unpin      (n/a)
        elpa       (n/a)
```
</pre></details>

**Comment by FollieHiyuki** [comment]
I don't think this is a bug. It's just not a sensible default for some people (including me, who prefers the default look of `org-tree-slide`)
It's a bit annoying that there is no way I can think of to revert the effect of some functions defined in `present.el`. Defining some variables to toggle those functions (hiding header, top level heading, ...) seems like a better option for me.

Anyway @ngharrison you can always install those packages manually (instead of inside `init.el`) and configure them manually.

**Comment by FollieHiyuki** [comment]
You can add this into your `config.el`
```
(advice-remove 'org-tree-slide--display-tree-with-narrow #'+org-present--narrow-to-subtree-a)
```

**Comment by FollieHiyuki** [comment]
> Content under subheadings is hidden and only the first subheading is shown. All the rest get skipped when moving to the next slide. I also can't move backwards in the slides, only forwards.

The default behavior is to hide the top heading, and cycle through subtree.
If you get annoyed by it, add this into your config:
```
(advice-remove 'org-tree-slide--display-tree-with-narrow
               #'+org-present--narrow-to-subtree-a)
```
> but then no images are shown

No image are show by default (not a bug). Inline image is only enabled with narrow hook. Look at the `present.el` file.

> no #+ blocks are removed

Yes `+org-present-hide-blocks-h` doesn't seem to work. You can take a look into `contrib-present.el` file to see what the problem is. I like to have those Org constructs so I don't care much.

> no symbols are prettified

This needs `+pretty` flag. And all it does is just changing heading, task indicators to nice symbols. I don't know what more need  prettified.
Look into `+org-present-prettify-slide-h` for details.

> Removing the config in ~/.emacs.d/modules/lang/org/contrib/present.el does revert the behavior back to something useable

Of course it does :) you made it come back to stock default settings of `org-tree-slide`

**Comment by ngharrison** [comment]
Maybe it was meant to never be used with subheadings. I can accept that even though I use it differently. But there are still a number of things, including basic ones, that don't seem to work as the present.el file indicates, or I just don't understand its purpose. It seems to inhibit the use of the original package. The obvious solution for me right now is to use my own configuration. I just bring this up in case anyone else is confused about the module as well.

> No image are shown by default (not a bug). Inline image is only enabled with narrow hook. Look at the present.el file.

I assumed showing inline images when it is narrowed is exactly what is done by `(add-hook 'org-tree-slide-mode-after-narrow-hook #'org-display-inline-images)`. I might misunderstand.

> This needs +pretty flag. And all it does is just changing heading, task indicators to nice symbols. I don't know what more need prettified.

By prettify I mean showing symbols (such as \rho) as single unicode characters and hiding *emphasis* markers. This is done without the +pretty flag and is performed by +org-pretty-mode, which is called inside `+org-present-prettify-slide-h`. With some further investigation, it looks like it toggles it instead of setting it. So if it was on before, it is turned off.

A few others. `org-tree-slide-content` doesn't show anything when only one level of headings is used and it has erratic behavior when multiple are. Once the end of the headings are reached (by using C-right), there is no way to go backwards (with C-left).

**Comment by h-bauer** [comment]
I also had recently issues with tree-slide-mode. I don't use images, only regular sections with text, bullet lists and sub-sections.

But when arriving on a subsection, the narrow did not occur as expected. The first sub-section was unfolded, the next two where still visible. I could not go further with the <C-Right> binding, unless moving the cursor to the next sibling. <C-Right> would then unfold the next sibling, but to move to the next section with <C-Right>, I had to move the cursor to the last line of the last sub-section.

After some investigation, I fixed my issue by removing the following advice :
```elisp
(after! org-tree-slide
  (advice-remove 'org-tree-slide--display-tree-with-narrow
                 #'+org-present--hide-first-heading-maybe-a)
  )
```

I could not fix it without the `after!` macro.

Hope this helps someone.

As a side note, I would have found helpful to have a variable to set the hard-coded margin defined in the autoload function `+org-present-prettify-slide-h`, via :
```elisp
(setq-local cwm-frame-internal-border 100)
```

**Comment by ngharrison** [comment]
Closing this since this was a general un-usability issue and the module has been updated and works much better in general.

**Comment by takaxp** [comment]
I'm the author of org-tree-slide. If you further want to add other comments on org-tree-slide, please visit https://github.com/takaxp/org-tree-slide/issues. Thanks.

**Comment by terinjokes** [comment]
I think it's re-opening this issue. The behavior still isn't documented, and it seems like an extreme departure from the upstream module? `C-right` moves forward three slides and then just stops advancing.

**Comment by ngharrison** [comment]
The module works for me now and I can't reproduce the behavior you describe. You should probably narrow down the exact problem and open a new issue.

**Comment by terinjokes** [comment]
It seems like the issue I ran into was the module unconditionally forcing the "simple" profile, rather than the default "presentation", so upstream's documentation didn't work as expected.

Copying the `+present` config without this forced profile selection and my presentations worked as expected. If Doom is going to set a profile, it would be nice to be able to disable it.
